datalab/datalab_session/tests/test_operations.py

```diff
@@ -0,0 +1,196 @@
+from django.test import TestCase
+
+from datalab.datalab_session.data_operations.data_operation import BaseDataOperation
+
+class SampleDataOperation(BaseDataOperation):
+    
+    @staticmethod
+    def name():
+        return 'SampleDataOperation'
+    
+    @staticmethod
+    def description():
+        return 'Testing class for DataOperation'
+    
+    @staticmethod
+    def wizard_description():
+        return {
+            'name': SampleDataOperation.name(),
+            'description': SampleDataOperation.description(),
+            'category': 'test',
+            'inputs': {
+                'input_files': {
+                    'name': 'Input Files',
+                    'description': 'The input files to operate on',
+                    'type': 'file',
+                    'minimum': 1,
+                    'maximum': 999
+                },
+                'input_number': {
+                    'name': 'Input Number',
+                    'description': 'The input number to operate on',
+                    'type': 'number',
+                    'minimum': 1,
+                    'maximum': 999
+                },
+                'input_string': {
+                    'name': 'Input String',
+                    'description': 'The input string to operate on',
+                    'type': 'string',
+                    'minimum': 1,
+                    'maximum': 999
+                }
+            }
+        }
+    
+    def operate(self):
+        self.set_output({'output_files': []})
+
+class TestDataOperation(TestCase):
+    def setUp(self):
+        input_data = {
+            'input_files': [],
+            'input_number': 1,
+            'input_string': 'test'
+        }
+        self.data_operation = SampleDataOperation(input_data)
```

Comment:
Why test a sample data operation and not one of the existing ones we have?

Reply:
I'm thinking this test class focuses on the basic shared functionality of all data operations using a sample, and the test classes for the specific operations focus just on their operate method. For example testing the generate_cache_key() here and not for each operation. That's my train of thought for this.